tendermint/src/lite/types.rs

```diff
@@ -71,19 +81,22 @@ pub trait Validator {
 /// It has an underlying Vote type with the relevant vote data
 /// for verification.
 pub trait Commit {
-    type Vote: Vote;
-
     /// Hash of the header this commit is for.
     fn header_hash(&self) -> Hash;
 
-    /// Return the underlying votes for iteration.
-    /// All votes here are for the correct block id -
-    /// we ignore absent votes and votes for nil here.
-    /// NOTE: we may want to check signatures for nil votes,
-    /// and thus use an ternary enum here instead of the binary Option.
-    // TODO figure out if we want/can do an iter() method here that returns a
-    // VoteIterator instead of returning a vec
-    fn into_vec(&self) -> Vec<Option<Self::Vote>>;
```

Comment:
The `into_vec` or any possibility to iterate over the underlying votes is not used in the abstract verifier code. The parts of the code that needed this is now moved into `voting_power_in` (see below). I think this makes the code more readable and closer to the spec.